# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/nsFontFaceLoader.cpp
# Commit: 93b7e171c74a
# Full Hash: 93b7e171c74a6aa88d98979009eb0e299e2b61a3
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-06-29 03:44:30
# Regressor Bug: 1771493
# File Overlap Count: 1
# Description:
#   Bug 1771493 - Part 1. Split FontFace(Set) into DOM facing and implementation classes. r=emilio
#   
#   This patch aims to split the implementation details of FontFace(Set)
#   into FontFace(Set)Impl classes. The Impl variants have threadsafe
#   refcounting and are intended to be the basis for worker support for
# ==============================================================================

diff -r c45dacd63805 -r 93b7e171c74a layout/style/nsFontFaceLoader.cpp
--- a/layout/style/nsFontFaceLoader.cpp	Tue Jun 28 14:12:34 2022 +0000
+++ b/layout/style/nsFontFaceLoader.cpp	Tue Jun 28 14:27:20 2022 +0000
@@ -45,7 +45,7 @@
 
 nsFontFaceLoader::nsFontFaceLoader(gfxUserFontEntry* aUserFontEntry,
                                    uint32_t aSrcIndex,
-                                   FontFaceSet* aFontFaceSet,
+                                   FontFaceSetImpl* aFontFaceSet,
                                    nsIChannel* aChannel)
     : mUserFontEntry(aUserFontEntry),
       mFontFaceSet(aFontFaceSet),
@@ -187,7 +187,7 @@
     nsTArray<gfxUserFontSet*> fontSets;
     ufe->GetUserFontSets(fontSets);
     for (gfxUserFontSet* fontSet : fontSets) {
-      nsPresContext* ctx = FontFaceSet::GetPresContextFor(fontSet);
+      nsPresContext* ctx = FontFaceSetImpl::GetPresContextFor(fontSet);
       if (ctx) {
         fontSet->IncrementGeneration();
         ctx->UserFontSetUpdated(ufe);
@@ -299,7 +299,7 @@
   nsTArray<gfxUserFontSet*> fontSets;
   mUserFontEntry->GetUserFontSets(fontSets);
   for (gfxUserFontSet* fontSet : fontSets) {
-    nsPresContext* ctx = FontFaceSet::GetPresContextFor(fontSet);
+    nsPresContext* ctx = FontFaceSetImpl::GetPresContextFor(fontSet);
     if (ctx) {
       // Update layout for the presence of the new font.  Since this is
       // asynchronous, reflows will coalesce.